types/beacon_types.go

```diff
@@ -231,16 +231,44 @@ func (n BeaconNetwork) EstimatedCurrentSlot() spec.Slot {
 
 // EstimatedSlotAtTime estimates slot at the given time
 func (n BeaconNetwork) EstimatedSlotAtTime(time int64) spec.Slot {
-	genesis := int64(n.MinGenesisTime())
-	if time < genesis {
+	// Sanitize time
+	if time < 0 {
 		return 0
 	}
-	return spec.Slot(uint64(time-genesis) / uint64(n.SlotDurationSec().Seconds()))
+
+	// Get delta time
+	genesis := n.MinGenesisTime()
+	deltaTime := uint64(time) - genesis
+
+	// Get slot duration
+	slotDurationInSeconds := int64(n.SlotDurationSec().Seconds())
+	// Sanitize slot duration
+	if slotDurationInSeconds <= 0 {
+		return spec.Slot(math.MaxUint64)
+	}
```

Comment:
I am wondering whether we need all of this in spec, especially since Slot Duration should be a constant...
GoSec yells if it isn't there?

Reply:
I agree. But, yes, GoSec yells (and with reason tbh 😅)